pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -517,3 +568,16 @@ func (s *NodeController) getCondition(node *api.Node, conditionType api.NodeCond
 	}
 	return nil
 }
+
+// latestConditionTime returns the latest condition timestamp for the node, regardless of condition status.
+// If nothing matches, the node creation timestamp will be returned.
+func (s *NodeController) latestConditionTime(node *api.Node, conditionType api.NodeConditionType) util.Time {
+	readyTime := node.ObjectMeta.CreationTimestamp
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == conditionType &&
+			condition.LastProbeTime.After(readyTime.Time) {
```

Comment:
In syncNodeStatus==true case, node controller will probe status from kubelet, so LastProbeTime makes sense here.  But in the 'false' case, it means the last time you got a resposne from kubelet, as you said.

Although it makes sense in the first case, I agree that LastStatusTime or LastConditionTime is a better, more general term for both cases.  I'm fine with the change, feel free to open an issue.  Let's see what Brian has in mind since he works on api mostly.